### PR TITLE
make built-in hooks

### LINKED ISSUE #260: ENH Add pre-commit hooks for supported tools to .pre-commit-hooks.yaml
Baiscally, this is how [pre-commit-hooks](https://github.com/pre-commit/pre-commit-hooks/blob/master/.pre-commit-hooks.yaml) does it.
This would allow to make the `.pre-commit-config.yaml` syntax a lot more concise.

Using:

```yaml
- repo: https://github.com/nbQA-dev/nbQA
  rev: 0.1.x
  hooks:
    - id: nbqa-flake8
    - id: nbqa-isort
```

Instead of:

```yaml
- repo: https://github.com/nbQA-dev/nbQA
  rev: 0.1.32
  hooks:
    - id: nbqa
      args: ["flake8"]
      name: nbqa-flake8
      alias: nbqa-flake8
      additional_dependencies: ["flake8"]
    - id: nbqa
      args: ["isort", "--nbqa-mutate"]
      name: nbqa-isort
      alias: nbqa-isort
      additional_dependencies: ["isort"]
```

Didn't test if it works, just leaving the idea here.
If adding the `args` to the `entry` doesn't work, we could always add additional entry points in `setup.py`.

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #260

### CODE REVIEW

File: docs/pre-commit.rst [hunk 1]
```diff
@@ -2,36 +2,32 @@
...
-            name: nbqa-isort
-            alias: nbqa-isort
-            additional_dependencies:
-            - isort
-
-See :ref:`configuration<configuration>` for how to further configure ``nbqa``.
+          args: [customtool]
+          name: nbqa-customtool
+          alias: nbqa-customtool
+          additional_dependencies [customtool]
```
Comment (girip11) [inline docs/pre-commit.rst:28]: `additional_dependencies: [customtool]`

**Review by girip11 — CHANGES_REQUESTED** [review]

File: docs/pre-commit.rst (hunk 1 above)
Comment (MarcoGorelli) [inline docs/pre-commit.rst:28]:
```suggestion
          additional_dependencies: [customtool]
```

Comment (MarcoGorelli) [reply docs/pre-commit.rst]: you're right, thanks for catching this!

File: docs/pre-commit.rst [hunk 2]
```diff
@@ -2,36 +2,32 @@
...
-            name: nbqa-flake8
-            alias: nbqa-flake8
-            additional_dependencies:
-            - flake8
+    - repo: https://github.com/nbQA-dev/nbQA
+      rev: 0.3.1
+      hooks:
+        - id: nbqa-black
+        - id: nbqa-isort
+        - id: nbqa-pyupgrade
```
[the remark this answers is not in the document]
Comment (girip11) [reply docs/pre-commit.rst:15]: This is an optional comment. I am approving the PR. Even if you fix this comment, feel free to merge the PR

[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply docs/pre-commit.rst:15]: That's a good suggestion - I've added a link to the pre-commit docs to avoid cluttering these too much

**Pull request merged**